src/Tests/Microsoft.NET.Sdk.WorkloadManifestReader.Tests/SdkDirectoryWorkloadManifestProviderTests.cs

```diff
@@ -31,6 +31,48 @@ void Initialize(string featureBand = "5.0.100", [CallerMemberName] string? testN
             Directory.CreateDirectory(_manifestVersionBandDirectory);
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ItShouldReturnTheWorkloadVersion(bool useWorkloadSet)
+        {
+            Initialize();
+
+            CreateMockManifest(_manifestRoot, "8.0.100", "ios", "11.0.2", true);
+            CreateMockManifest(_manifestRoot, "8.0.200", "android", "33.0.2-rc.1", true);
+            CreateMockManifest(_manifestRoot, "8.0.200-rc.2", "maui", "15.0.1-rc.456", true);
+
+            if (useWorkloadSet)
+            {
+                CreateMockWorkloadSet(_manifestRoot, "8.0.200", "8.0.200", @"
+                    {
+                    ""ios"": ""11.0.2/8.0.100"",
+                    ""android"": ""33.0.2-rc.1/8.0.200"",
+                    ""maui"": ""15.0.1-rc.456/8.0.200-rc.2"",
+                    }
+                ");
+            }
+
+            var sdkDirectoryWorkloadManifestProvider
+                = new SdkDirectoryWorkloadManifestProvider(sdkRootPath: _fakeDotnetRootDirectory, sdkVersion: "8.0.200", userProfileDir: null, globalJsonPath: null);
+
+            if (useWorkloadSet)
+            {
+                sdkDirectoryWorkloadManifestProvider.GetWorkloadVersion().Should().Be("8.0.200");
+            }
+            else
+            {
+                string[] manifests = sdkDirectoryWorkloadManifestProvider.GetManifests().OrderBy(m => m.ManifestId).Select(m => $"{m.ManifestId}.{m.ManifestFeatureBand}.{m.ManifestVersion}").ToArray();
+                manifests.Length.Should().Be(1);
+                manifests.Should().Contain("android.8.0.200.33.0.2-rc.1");
+                sdkDirectoryWorkloadManifestProvider.GetWorkloadVersion().Should().Be("8.0.200-manifests.4ba11739");
+            }
+
+            Directory.Delete(Path.Combine(_manifestRoot, "8.0.100"), recursive: true);
+            Directory.Delete(Path.Combine(_manifestRoot, "8.0.200"), recursive: true);
+            Directory.Delete(Path.Combine(_manifestRoot, "8.0.200-rc.2"), recursive: true);
```

Comment:
It shouldn't be necessary to delete anything here, each test should use a test-specific directory.